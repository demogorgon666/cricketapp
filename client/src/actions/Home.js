import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
