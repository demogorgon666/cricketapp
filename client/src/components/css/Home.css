@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

* {
  font-family: Montserrat, sans-serif;
}

.div-container {
  margin-top: 80px;
}

.div-news {
  width: 100%;
  height: 140px;
  border: 1px solid #707070;
  border-radius: 8px;
  margin-bottom: 30px;
  display: flex;
}
/* for news images */
.div-recent-news {
  width: 100%;
  border: 1px solid #707070;
  border-radius: 8px;
  margin-bottom: 30px;
  display: flex;
}

.div-second-section {
  width: 30vw;
}

.div-news-details {
  width: 80%;
  text-decoration: none;
  margin: 2%;
}

/* .news_img {
	width: 100%;
	height: 100%;
} */
.div-news-images {
  border-radius: 8px 0px 0px 8px;
  width: 40%;
  display: inline-block;
  /* height: 100%; */
  /* border-radius: 50%; */
  background-repeat: no-repeat;
  background-position: center center;
  background-size: cover;
}

.div-news-section {
  padding: 20px 60px 20px 100px;
  width: 60vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.div-section {
  display: flex;
}

.p-news {
  color: #272727;
  font-size: 20px;
  margin: 0px;
}
.home-news-date {
  font-size: 12px;
}
.link-news {
  text-decoration: none;
}
/* --------------------------------- */
