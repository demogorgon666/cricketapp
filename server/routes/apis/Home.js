const express = require("express");
const router = express.Router();
const pg = require("pg-promise")();
const db = pg("postgres://postgres:root@localhost/cricketalpha");

router.get("/news", async (req, res) => {
	const news_result = await db.any("Select * from news");
	for (news of news_result) {
		var date = new Date(news.news_date);

		news.news_date = date.toLocaleDateString("en-IN", {
			weekday: "short",
			month: "long",
			day: "2-digit",
			year: "numeric"
		});
	}
	// console.log(news_result);
	res.status(200).json({
		status: 200,
		data: news_result,
		message: "Retrieved all news"
	});
	// console.log(news_result);
});
module.exports = router;
