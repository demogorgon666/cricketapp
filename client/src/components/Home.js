import React, { Component } from "react";
import { connect } from "react-redux";
import { getNews, getRecentMatches } from "../actions/Home";
import "./css/Home.css";
import news_img from "../components/images/vk.jpeg";
import logo from "./images/indialogo.jpg";
import { getRanks } from "../actions/Teams";
// import { getRanks } from "../actions/Teams";
import MostWins from "./MostWins";
import { Link } from "react-router-dom";

export class Home extends Component {
  componentWillMount() {
    console.log("home mounted");
    this.props.getNews();
    console.log(this.props);
    // this.props.getTeams();
    let ranking = {
      match_type: "Test",
    };
    this.props.getRanks(ranking);
    // this.props.getNews();
  }
  onClickTest = e => {
    e.preventDefault();
    let ranking = {
      match_type: "Test",
    };
    this.props.getRanks(ranking);
  };

  onClickT20 = e => {
    e.preventDefault();
    let ranking = {
      match_type: "T20",
    };
    this.props.getRanks(ranking);
  };

  onClickOdi = e => {
    e.preventDefault();
    let ranking = {
      match_type: "ODI",
    };
    this.props.getRanks(ranking);
  };
  render() {
    return (
      <div className="div-container">
        <div className="div-section">
          <div className="div-news-section">
            {this.props.home.map(news => (
              <Link
                className="link-news"
                to={{
                  pathname: "/newsbyid/" + news.news_id,
                  // state:{
                }}
              >
                <div className="div-news">
                  <div
                    className="div-news-images"
                    style={{
                      backgroundImage: `url(${news_img})`,
                    }}
                  >
                    {/* <img className="news_img" src={news_img} /> */}
                  </div>
                  <div className="div-news-details">
                    <p className="p-news">
                      <b>{news.news_title}</b>
                    </p>
                    <p className="home-news-date">{news.news_date} </p>
                  </div>
                </div>
              </Link>
            ))}
          </div>
          <div id="home-recent-matches" />
          {/* ------------------------------------- */}
          <div className="div-second-section">
            <MostWins />
          </div>
          {/* -------------------------------------- */}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  home: state.HomeReducer.home,
  ranks: state.TeamsReducer.ranks,
});
export default connect(
  mapStateToProps,
  {
    getNews,
    getRanks,
    getRecentMatches,
  }
)(Home);
