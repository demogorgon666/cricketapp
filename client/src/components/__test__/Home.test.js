import React from 'react'
import { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// import { getNews } from "../../actions/Home";
// import { getRanks } from "../../actions/Teams";

import { Home } from '../Home'

// const home = jest.fn();
// const onClickOdi = jest.fn();
const getNews = jest.fn()
const getRanks = jest.fn()
const home = []
const ranks = []

const wrapper = shallow(
  <Home
    getNews={getNews}
    getRanks={getRanks}
    home={home}
    ranks={ranks}
    // onClickOdi={onClickOdi}

    // singlePlayer={singlePlayer}
    // match={{ isExact: true, params: { path: "/", url: "/" } }}
  />
)
describe('Testing for home component', () => {
  it('should render the component', () => {
    // const wrapper = mount(<Home home={home} />);
    expect(wrapper).toMatchSnapshot()
  })

  it('should display the two teams' , () => {
    
  })
  // it("should have 3 div tags with className as 'p-card'", () => {
  //   expect(wrapper.find(".p-card").length).toBe(3);
  // });

  // Made the component reuseable

  // it("should have 3 divs under p-card named as Test, T20 and ODI", () => {
  //   expect(
  //     wrapper
  //       .find(".p-card")
  //       .at(1)
  //       .text()
  //   ).toBe("ODI");
  // });
  // it("should hav 3 divs under p-card named as Test, T20 and ODI", () => {
  //   expect(
  //     wrapper
  //       .find(".p-card")
  //       .at(2)
  //       .text()
  //   ).toBe("T20");
  // });
  // it("should hav 3 divs under p-card named as Test, T20 and ODI", () => {
  //   expect(
  //     wrapper
  //       .find(".p-card")
  //       .at(0)
  //       .text()
  //   ).toBe("Test");
  // });

  // it("should have 3 divs under p-card named as Test, T20 and ODI", () => {
  //   expect(
  //     wrapper
  //       .find(".p-topteam")
  //       .at(0)
  //       .text()
  //   ).toBe("Most Wins");
  // });
  // it("should consists in total 12 div tags", () => {
  //   expect(wrapper.find("div").length).toBe(12);
  // });
})
