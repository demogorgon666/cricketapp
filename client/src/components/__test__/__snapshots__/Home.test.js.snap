// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for home component should render the component 1`] = `
<div>
  <Connect(Navbar)
    changeGender={[Function]}
  />
  <div
    className="div-container"
  >
    <div
      className="div-section"
    >
      <div
        className="div-news-section"
      >
        <Link
          className="link-news"
          to={
            Object {
              "pathname": "/newsbyid/1",
            }
          }
        >
          <div
            className="div-news"
          >
            <div
              className="div-news-images"
              style={
                Object {
                  "backgroundImage": "url(vk.jpeg)",
                }
              }
            />
            <div
              className="div-news-details"
            >
              <p
                className="p-news"
                id="p-news"
              >
                <b>
                  this is title
                </b>
              </p>
              <p
                className="home-news-date"
                id="home-news-date"
              >
                12-12-12
              </p>
            </div>
          </div>
        </Link>
      </div>
      <div
        id="div-second-section"
      >
        <div
          id="home-recent-matches"
        >
          <div
            id="home-recent-matches-title"
          >
            Recent Matches
          </div>
          <Link
            id="link-recent-matches"
            to={
              Object {
                "pathname": "/match/details/1",
              }
            }
          >
            <div
              id="home-match"
            >
              <div
                id="team-score"
              >
                <p
                  id="home-recent-matches-teamOne"
                >
                  India
                </p>
                <p
                  id="home-recent-matches-team-one-score"
                >
                  120
                  /
                </p>
              </div>
              <div
                id="team-score"
              >
                <p
                  id="home-recent-matches-teamTwo"
                >
                  Pakistan
                   
                </p>
                <p
                  id="home-recent-matches-team-two-score"
                >
                  150
                  /
                   
                   
                </p>
              </div>
              <div
                id="home-recent-matches-result"
              >
                <p
                  id="home-recent-matches-team-winner"
                >
                  Pakistan
                   
                   
                </p>
              </div>
              <div
                id="home-recent-match-date"
              >
                12-12-12
              </div>
            </div>
          </Link>
        </div>
        <Connect(MostWins) />
      </div>
    </div>
  </div>
</div>
`;
